File: go/all-your-base/all_your_base.go
package allyourbase

import (
    "math"
    "errors"
)

func ConvertToBase(inputBase int, inputDigits []int, outputBase int) ([]int, error) {
	if inputBase < 2 {
        return nil, errors.New("input base must be >= 2")
    }
	if outputBase < 2 {
        return nil, errors.New("output base must be >= 2")
    }
    baseTenNumber := 0
    for i := 0; i < len(inputDigits); i++ {
        if inputDigits[i] < 0 || inputDigits[i] >= inputBase {
            return nil, errors.New("all digits must satisfy 0 <= d < input base")
        }
        baseTenNumber += inputDigits[i] * int(math.Pow(float64(inputBase), float64(len(inputDigits) - i - 1)))
    }
	index := 0
	for i := 1; int(math.Pow(float64(outputBase), float64(i))) <= baseTenNumber; i++ {
        index += 1
    }
	output := []int{}
    for i := index; i >= 0; i-- {
        digit := baseTenNumber / int(math.Pow(float64(outputBase), float64(i)))
        output = append(output, digit)
        baseTenNumber -= digit * int(math.Pow(float64(outputBase), float64(i)))
    }
	return output, nil
}

File: python/all-your-base/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError('input base must be >= 2')
    if output_base < 2:
        raise ValueError('output base must be >= 2')
    number = 0
    for index, value in enumerate(digits[::-1]):
        if value < 0 or value >= input_base:
            raise ValueError('all digits must satisfy 0 <= d < input base')
        number += value * input_base ** index
    if number == 0:
        return [0]
    new_digits = []
    while number > 0:
        new_digits.insert(0, (number % output_base))
        number //= output_base
    return new_digits
